### ISSUE TITLE
Adjusting tile size?

### ISSUE THREAD

**Issue by risoldi** [arscan/encom-globe#9]
Hi again,

I'm almost there, I adapted the colors and the fog in order to have a grayscale globe which fades to transparent in the background. My last issue would be that, as it is, the empty space between the tiles is a bit too much. What I'd like to do is increase the size of the tiles, so that they fill the space some more. I tried fiddling with the code a little, but I can't find where I need to tweak (admittedly, I am a complete noob when it comes to WebGL and 3D programming in general).
From what I understood, the surface is a mesh of triagles and something somehow (the createParticles function, I believe?) makes the tiles on top of that. Do you think this is feasible by changing some parameters somewhere, or would it involve changing the mesh and thus redoing all calculations and rewriting major portions of the code?

**Comment by arscan** [comment]
Yeah, so you would like the tiles to look more like the example I have here, right?

http://www.robscanlon.com/hexasphere/

I originally had the "hexasphere" generated at page load time in Encom Globe using that library, so you could pass in parameters that defined the number, size, etc of the tiles to encom-globe.  But unfortunately I found that it is fairly CPU intensive, and that caused pages to lock for a few seconds while the tiles were calculated. So instead I decided to have the tiles be generated ahead of time, and be included in a file that sets a global variable that Encom Globe picks up (see grid.js).  

You can regenerate grid.js with the right parameters for you using the nodejs script buildgrid in the bin directory.  Try this:

```bash
bin/buildgrid -r 500 -o grid.js -m resources/equirectangle_projection.png -s 1.0
```

The -s flag is what changes the "scale" of a given tile.  A "scale" of 1.0 will completely fill the tile up in its available space.  A "scale" of .45 (which is the default), will give it a good amount of buffer.  It looks like the script also adds a certain amount of randomness to the scale as well, which you might want to hack out.  Again, this was built to mimic the movie, so it isn't particularly user friendly if you want to give it a different look.  Theoretically the output of that script will be a new grid.js file, which you can replace the one that comes with this.

Let me know how that goes.   There is a decent chance it won't work, as I may have broken something in that script since I first made this -- once I had a set of tiles that looked good I basically neglected this part of the code.

**Comment by risoldi** [comment]
Hey, thanks for the pointers. Good news is, the script still works! :) 
So, in order to help other people who may want to do this, here's what I did:
- got the hexasphere and encom-globe projects from GitHub, then unzipped them;
- took the js files from the src directory of hexasphere and placed them in the encom-globe-master directory;
- installed node.js and the following packages for node: minimist, pngjs, and fs
- modified the bin/buildgrid script in order to hardcode the size = 1.0 and the path to the map and the hexasphere.js file (I only needed this once, so I did not fiddle with passing parameters)
- I modified the line that says:
```javascript
var scale = size - Math.random() * .25;
```
to read just: 
```javascript
var scale = size;
```
- Finally, I ran:
```
bin/build -o grid.js
```

And voilà, I have my globe with no gaps between hexagons. :)
